app/src/test/java/org/simple/clinic/appconfig/AppConfigRepositoryTest.kt

```diff
@@ -290,4 +292,24 @@ class AppConfigRepositoryTest {
     val fetchedStatesResult = repository.fetchStatesInSelectedCountry()
     assertThat(fetchedStatesResult).isEqualTo(StatesResult.FetchError(ResolvedError.Unexpected(cause)))
   }
+
+  @Test
+  fun `saving the state, should save it to local persistence`() {
```

Comment:
Hmm, I feel this verification is more suited to an instrumented test rather than a unit test since it actually verifies the saving instead of a method call.

Reply:
Yeah, I tried to keep it consistent with existing `AppConfigRepository` tests. Like we discussed will create an integration test, would that be fine?

Reply:
Along with the API integration test, I think we can move this entire test to be an integration test as well, wdyt?

Reply:
That works.